uint/src/uint.rs

```diff
@@ -1307,6 +1301,34 @@ macro_rules! construct_uint {
 
 				$name(ret)
 			}
+
+			fn fmt_hex(&self, f: &mut $crate::core_::fmt::Formatter, is_lower: bool) -> $crate::core_::fmt::Result {
+				let &$name(ref data) = self;
+				// special case.
+				if self.is_zero() {
+					return f.pad_integral(true, "0x", "0");
+				}
+
+				let mut latch = false;
+				let mut buf = String::new();
```

Comment:
hmm, this doesn't compile for no_std
why do we use a string and not directly write to f as before?

Reply:
This is because `pad_integral` for padding requires constructed `str`.
If I implement `pad_integral` manually, `String` can be removed probably.
I'll try it.